src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,124 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+// class Ui - handles all the lines to be printed and all the user inputs (commands)
+public class Ui {
+
+    public Ui() {
+
+    }
+
+    // prints out welcome greeting to user when Duke is run
+    public void showWelcome() {
+        System.out.println("    Hi! I'm Samantha\n    How can I help?");
+    }
+
+    public void takeCommands(TaskList t) {
+        Scanner sc = new Scanner(System.in);
+        String s = sc.nextLine();
+
+        ArrayList<Task> tasks = t.getTasks();
+        int taskCounter = tasks.size();
+        while (!s.equals("bye")) {
+            try {
+                // method handleCommand handles current command, returns updated taskCounter variable
+                taskCounter = handleCommand(s, tasks, taskCounter);
```

Comment:
What is the rationale of keeping track of a taskCounter variable instead of using the builtin `list.size()` method?